modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/exp/RexToLixTranslator.java

```diff
@@ -214,7 +218,18 @@ Expression translateCast(
         Expression convert = null;
         switch (targetType.getSqlTypeName()) {
             case ANY:
-                convert = operand;
+                // IgniteCustomType: conversion from some type to possibly a custom data type.
+                // Should we add a hook here that can short-circuit when a custom type can not be converted?
+                if (targetType instanceof UuidType) {
+                    // We need to convert an argument to an object so a call will throw a CastCastException
+                    // instead of a NoSuchMethodError in runtime.
+                    // It would be even better if this cast were not be necessary.
+                    return Expressions.call(UuidFunctions.class, "cast", Expressions.convert_(operand, Object.class));
```

Comment:
Can it be encapsulated into customtype class, to reduce places that require modification?

Reply:
I can introduce UuidFunctions.CAST (static field) and replace `Expressions.call(UuidFunctions.class, "cast", operand_expression)` with Expressions.call(UuidFunctions.CAST, operand_expression). I think it is better to keep explicit code here - we want to build a cast expression with such and such operand.

Reply:
I mean have abstract method in IgniteCustomType and realization like a 
`public Expression cast() {
return Expressions.call(UuidFunctions.class, "cast", Expressions.convert_(operand, Object.class));
`

Reply:
If you look at the code more carefully one part does not issue a cast to an object. And that's a huge distinction.

Reply:
I think I misunderstood you. I definitely going to do that.